### PR TITLE
Add documentation about environment variables

### PR DESCRIPTION
**Pull request by henrycatalinismith**
Talked about documenting this in https://github.com/zetkin/app.zetkin.org/pull/2447. Here's a first draft!

| environment-variables.md |  ClientContextEnvVars | ScaffoldedEnvVars |
|-|-|-|
| 127 0 0 1_8081_documents_Environment_Variables html |  127 0 0 1_8080_types_ClientContextEnvVariables html (3) | 127 0 0 1_8080_types_ScaffoldedEnvVars html (1) |

### CODE REVIEW

File: src/utils/next.ts [hunk 1]
```diff
@@ -26,7 +26,25 @@ type RegularProps = {
...
+ */
+export type ScaffoldedEnvVars = {
+  FEAT_AREAS: string | null;
+  MUIX_LICENSE_KEY: string | null;
+  ZETKIN_APP_DOMAIN: string | null;
+  ZETKIN_GEN2_ORGANIZE_URL: string | null;
+};
+
 export type ScaffoldedProps = RegularProps & {
+  envVars: ScaffoldedEnvVars;
```
Comment (richardolsson) [inline src/utils/next.ts:47]: Wow I can't believe this was missed. I'm glad you caught it!

In ba86abc3c963a8f6b91cae4932382e69437d9878 (part of #2447) I made some minor changes and used `Environment['vars']` as the type. I don't particularly care about the name of the type, and I completely appreciate that using [indexed access types](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html) is not everyone's preference, or perhaps even mine?

So if you want to change all of these places to use a shared, named type instead that is good IMO!

File: typedoc.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "cleanOutputDir": true,
   "name": "Zetkin",
   "entryPoints": ["./docs/index.ts"],
-  "out": "./typedoc",
+  "out": "./public/docs",
```
Comment (richardolsson) [inline typedoc.json:8]: Much better than passing this particular option in the CLI.

File: docs/environment-variables/environment-variables.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+environments per build, where environment variables cannot be known at build
+time. It also happens to be Zetkin's deployment model.
+
+The emphasis that Next.js places on the Vercel platform's deployment
+model in its architecture and documentation can easily lead to confusion when
+working on an app like Zetkin with a different deployment model. The fact that
+handling runtime environment variables looks different under the newer app
+router paradigm compared to the older pages router code also adds to the sense
+of complexity here. So here's how Zetkin's runtime environment variables work in
+both Next.js routing paradigms.
```
Comment (richardolsson) [inline docs/environment-variables/environment-variables.md:31]: Love this!

**Review by richardolsson — CHANGES_REQUESTED** [review]
I think this is a great addition to the docs. I'm sorry for pulling the rug out on this otherwise (almost) mergable PR but I made some changes in #2447 before merging it that are now causing merge conflicts (and perhaps worse) here.

Basically, the types for environment variables should be consistent across the pages and app router. I took one small step in that direction in ba86abc3c963a8f6b91cae4932382e69437d9878 but maybe this is a good PR to make it fully consistent?

**Comment by henrycatalinismith** [comment]
Oh nice, glad you aligned these already. I think it will make adding and removing these things even easier as you'll be able to update one central type and then treat the resulting type errors as a to-do list for completely integrating the new env var into both the app router and the page router. Looks like only teeny tiny changes ahead here so with any luck won't take long if I get a few spare minutes here and there.

**Comment by henrycatalinismith** [comment]
Right then, the  merge commit is https://github.com/zetkin/app.zetkin.org/pull/2452/commits/0e1a520520a1b1700de1ce3e339c7cc9d19d7e6d.

At first I was going to sit and daydream about a nice general name for a shared type. Then it occurred to me that maybe this isn't something that needs to be so meticulously neutral. If the goal state is to one day have everything be on the app router then maybe just focusing on the app router a little bit in these kinds of details is best. So I kept `ClientContextEnvVars`, added everything to that, and imported it in `scaffold()` to reuse there. What do you reckon?

**Comment by henrycatalinismith** [comment]
Should probably add something about useEnv to this right? Adding a new env var is covered nicely here but your other most likely question as someone who maybe knows Next.js but is new to Zetkin is how to read them.

**Comment by henrycatalinismith** [comment]
lol I just found another `EnvVars` type in [src/core/env/Environment.tsx](https://github.com/zetkin/app.zetkin.org/blob/main/src/core/env/Environment.ts). It's actually kind of cool how documenting this stuff ends up being a way to take a holistic look at it and sometimes find ways of tidying it up.

Do we want to use that instead of this new ClientContextEnvVars? The fact that it makes everything optional seems to be really convenient for writing stuff like unit tests where I see lots of places defining only the ones that matter for the individual test.

**Comment by richardolsson** [comment]
> lol I just found another `EnvVars` type in [src/core/env/Environment.tsx](https://github.com/zetkin/app.zetkin.org/blob/main/src/core/env/Environment.ts). It's actually kind of cool how documenting this stuff ends up being a way to take a holistic look at it and sometimes find ways of tidying it up.
> 
> Do we want to use that instead of this new ClientContextEnvVars? The fact that it makes everything optional seems to be really convenient for writing stuff like unit tests where I see lots of places defining only the ones that matter for the individual test.

Yeah that sounds like a good idea! But unless I'm thinking about this the wrong way, it will mean that you lose the benefit of being able to change the type and get the type errors as a Todo.

I guess you can change the type without making the new var optional, handle the errors, and then make it optional.

**Review by richardolsson — APPROVED** [review]
Nice! I think this is ready to merge! 💯 

Let me just say again, lovely to have you back on the roster! 🙌

**Pull request merged**